Sources/SwiftInspectorVisitors/NestableTypeInfo.swift

```diff
@@ -22,26 +22,38 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
-public struct ClassInfo: Codable, Equatable {
+protocol NestableTypeInfo {
+
+  init(
```

Comment:
Do we need this initializer in the protocol? I usually feel like it's a bit of a smell to have initializers in the definition of protocols since it's no longer saying what this type is, but it's being prescriptive about implementation details. If we can remove it I'd vote for that